daemon/kmd/wallet/driver/ledger.go

```diff
@@ -297,3 +336,17 @@ func (lw *LedgerWallet) signTransactionHelper(tx transactions.Transaction) (sig
 	copy(sig[:], reply)
 	return
 }
+
+func (lw *LedgerWallet) signProgramHelper(data []byte) (sig crypto.Signature, err error) {
+	lw.mu.Lock()
+	defer lw.mu.Unlock()
+
+	// TODO: this is probably wrong and needs extending the client code on the device
+	reply, err := lw.dev.Exchange(data)
```

Comment:
The ledger expects data in a particular format -- I think this method should just return an error for now if the ledger firmware isn't ready.

Reply:
ok, errors out now instead of doing something nonsensical